Update to Python 3.12

jmacario24 opened the pull request:
Update Python to version 3.12 and dependencies.
Change to venv execution. Add Makefile for
ease of development.

This resolves MON-13564.

jmacario24 committed:
Update to Python 3.12

Update Python to version 3.12 and dependencies.
Change to venv execution. Add Makefile for
ease of development.

This resolves MON-13564.

Signed-off-by: Jerick Macario <jmacario@itrsgroup.com>

jdumalaonITRS commented on op5build/check_k8s.spec:
```diff
@@ -1,5 +1,8 @@
 %define exec_path check_k8s.py
-%define pkg_path k8s
+%define app_install_path /opt/monitor/op5/check_k8s
```
You might want to rethink the dir strucure here. 
If we include all python plugins as venv and use /opt/monitor/op5 as install path, this dir could get cluttered

jmacario24 committed:
Restructure app location

Restructure app locations to plugins directory.

Part of MON-13564.

Signed-off-by: Jerick Macario <jmacario@itrsgroup.com>

jmacario24 committed:
Update shebang for portability

Update main script shebang line to make it
portable. Build will replace to the python path.

Part of MON-13564.

Signed-off-by: Jerick Macario <jmacario@itrsgroup.com>

jmacario24 commented on check_k8s.py:
```diff
@@ -1,76 +1,6 @@
-    msg = NAGIOS_MSG.format(state=output.state.name, message=output.message)
-    output.channel.write(msg)
-    sys.exit(output.state.value)
+#!/opt/plugins/check_k8s/bin/python -I
```
I've updated to keep this on the shebang line. This will now be replaced by the mangling through the defined value in the spec file.
